gc/base/SweepPoolManagerAddressOrderedListBase.hpp

```diff
@@ -47,16 +47,14 @@ class MM_HeapLinkedFreeHeader;
 class MM_SweepPoolManagerAddressOrderedListBase : public MM_SweepPoolManager
 {
 private:
-
 protected:
+public:
 
+private:
+protected:
 	MMINLINE void calculateTrailingDetails(MM_ParallelSweepChunk *sweepChunk, uintptr_t *trailingCandidate, uintptr_t trailingCandidateSlotCount);
 	MMINLINE virtual void connectChunkPostProcess(MM_ParallelSweepChunk *chunk, MM_SweepPoolState *sweepState, MM_HeapLinkedFreeHeader* splitCandidate, MM_HeapLinkedFreeHeader* splitCandidatePreviousEntry){}
 	MMINLINE virtual void addFreeMemoryPostProcess(MM_EnvironmentBase *env, MM_MemoryPoolAddressOrderedListBase *memoryPool, void *addrBase, void *addrTop, bool needSync, void *oldAddrTop = NULL) {}
-	MMINLINE virtual bool isEligibleForFreeMemory(MM_EnvironmentBase *env, MM_MemoryPoolAddressOrderedListBase *memoryPool, void* address, uintptr_t size)
-	{
-		return memoryPool->canMemoryBeConnectedToPool(env, address, size);
-	}
```

Comment:
It's probably ok to still use a simple helper (if anything, to easier spot the code that relies on min free size check). It would not be virtual and no need to pass env and pool info (which would possibly require to rename it slightly?)

Reply:
new private method 	MMINLINE bool isFreeSizeEligibleForRecycling(uintptr_t size, uintptr_t minFreeSize)

Reply:
actually, it would even cleaner if the helper is in Chunk class , and the call would look something like 

chunk->isFreeSizeEligibleForRecycling(leadingFreeEntrySize)

Reply:
you can even have 2 methods, one general, where you pass the size and another specialized where it assumes the check against trailingFreeCandidateSize of the same chunk